spec/formatting.md

```diff
@@ -219,15 +229,24 @@ the following steps are taken:
    as long as reasonable precautions are taken to keep the function interface
    simple and minimal, and avoid introducing potential security vulnerabilities.
 
-   As implementations MAY allow custom functions to be defined by users,
-   their access to the _formatting context_ SHOULD be minimal and read-only,
-   and their execution time SHOULD be limited.
+   An implementation MAY define its own functions.
+   An implementation MAY allow custom functions to be defined by users.
 
-5. If the call succeeds,
+   An implementation MAY perform additional processing before calling the function.
+   For example, it can modify the _name_ of an _option_ to match the function's
+   input expectations (such as by removing the _namespace_ prefix from the _name-part_).
+
```

Comment:
This feels a bit weird, and I don't think it should be allowed. The functions that an implementation will end up calling should be considered rather specific to that implementation, and therefore fully capable of dealing with their options without preprocessing.

For example, let's say an implementation provides an `icu:number` function that's mostly the same except for the core `number` but it adds a few options. That function may internally support both `icu:skeleton` and `skeleton` as synonyms for each other, but figuring that out ought to be on the function, rather than the implementation. E.g. detecting that both should not be defined at the same time ought to be a function concern.

```suggestion
```

Reply:
I think of the function and the implementation of the function interchangeably. This permits an implementation to say to plug-in functionality that the namespacing will be managed for the function. But we could leave it out.